src/dex-ui/services/DexService/dao/DAOService.ts

```diff
@@ -245,8 +280,23 @@ async function fetchNFTDAOs(eventTypes?: string[]): Promise<NFTDAODetails[]> {
     new ethers.utils.Interface(FTDAOFactoryJSON.abi),
     eventTypes
   );
-  return logs.map((log): NFTDAODetails => {
+  const allPromises = logs.map(async (log): Promise<NFTDAODetails> => {
     const argsWithName = getEventArgumentsByName<NFTDAOCreatedEventArgs>(log.args, ["webLinks"]);
+
+    argsWithName.daoAddress = (await DexService.fetchContractId(argsWithName.daoAddress)).toSolidityAddress();
```

Comment:
Create new variables instead of mutating the existing object (`argsWithName`). The async mutations can lead to issues that are difficult to trace and debug.